### ISSUE TITLE
Plain HTTP services being identified as ssl/http instead of http by https-redirect

### ISSUE THREAD

**Issue by djcater** [nmap/nmap#1400]
If I checkout the latest code from Github and run:
```
sudo nmap -4 -vvv -d -sS -p80 -sV -sC scanme.nmap.org
```
The scan takes longer than it did previously, and I can see default NSE SSL scripts running such as ssl-date, ssl-known-key, ssl-cert etc. and then in the output port 80 is identified as "ssl/http" and errors such as from ssl-date: "ERROR: Unable to obtain data from the target".

On larger scans this is causing hundreds of script threads to hold up the scan until they timeout against non-SSL services.

If I use `--version-trace` both scans print:
```
Service scan match (Probe GetRequest matched with GetRequest line xxx): 45.33.32.156:80 is http.
```
But the current code then labels the service as `ssl/http` instead of `http`.

@bonsaiviking Is this perhaps related to 9c3e6768714b09d3d9aeaa63f47cda14d5019991 or 7a790d36656896e7fee42553180269a702e58c22?

**Comment by djcater** [comment]
Comparing a self-built Nmap 7.70SVN from the latest code against Nmap 7.60 (that's what I have to compare against at the moment in Ubuntu 18.04), the difference is 8 seconds versus 1 minute 10 seconds.

This means the this one-port scan is taking over 7 times as long!

**Comment by djcater** [comment]
Maybe it's not related to those nselib changes, as the service is identified as `ssl/http` just on the basis of `-sV` without `-sC`.

**Comment by djcater** [comment]
Found the answer!

The `https-redirect` script (introduced in June in cd3253f5a2d5e7bd42c23ed24fe7c378feafc218) is a default `version` script, and for some reason it is identifying the service as `ssl`.

I can reproduce this by running and comparing:
```
sudo nmap -4 -vvv -d -sS -p0 scanme.nmap.org
sudo nmap -4 -vvv -d -sS -p0 --script="https-redirect" scanme.nmap.org
```
I worked around this by renaming the `https-redirect.nse` file to `https-redirect.nse.old` in `/usr/local/share/nmap/scripts/` and now the scan finishes quicker and doesn't identify the service as `ssl/http`.
